util/generic/simple_span.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+template <class T>
+class TSimpleSpan {
+private:
+    T* data_;
+    size_t len_;
+public:
+    TSimpleSpan(size_t len): len_(len) {
+        data_ = new T[len];//(T*)y_allocate(len*sizeof(T));
+    }
+    TSimpleSpan(): data_(nullptr), len_(0) {
+    }
+    ~TSimpleSpan() {
+        if (data_) {
+            delete [] data_;//y_deallocate(data_);
+        }
+        data_ = nullptr;
+        len_  = 0;
+    }
+    TSimpleSpan (const TSimpleSpan &) = delete;
+    TSimpleSpan& operator=(const TSimpleSpan& other) = delete;
+    TSimpleSpan (const TSimpleSpan &&) = delete;
+    TSimpleSpan& operator=(const TSimpleSpan&& other) = delete;
+
+    void reset(T* data, size_t len) {
+        if(data_) {
+            delete [] data_;
+        }
+        data_ = data;
+        len_ = len;
+    }
+    inline T* data() const {
+        return data_;
+    }

```

Comment:
please use TVector and TArrayRef, TConstArrayRef instead

Reply:
done, TArrayRef was used instead